openapi.yaml

```diff
@@ -0,0 +1,154 @@
+                content:
+                $ref: '#/components/schemas/to-do'
+        '500':
+          description: server error
+  /to-dos/{to-doID}:
+    get:
+      summary: Returns a to-do by ID.
+      parameters: 
+        - name: to-doID
+          in: path
+          required: true
+          description: Parameter of to-do that you wanted the display
+          schema:
+            $ref: '#/components/schemas/to-do'
+      responses:
+        '200':
+          description: Parameters Displayed
+          content:
+            application/json:
+              schema:
+                type: string
+                content:
+                $ref: '#/components/schemas/to-do'
+        '500':
+          description: server error
+    delete:
+      summary: Delete a to-do by ID
+      parameters: 
+        - name: to-doID
+          in: path
+          required: true
+          description: Parameter of to-do that you wanted the display
+          schema:
+            $ref: '#/components/schemas/to-do'
+      responses:
+        '200':
+          description: Json object deleted
+          content:
+            application/json:
+              schema:
+                type: string
+                content:
+                $ref: '#/components/schemas/to-do'
+        '500':
+          description: server error
+  /to-dos/{to-doDetail}:
+    patch:
+      summary: Update to-do Detail
+      parameters:
+        - name: to-doDetail
+          in: path
+          required: true
+          description: Parameter of to-do that you wanted the update
+          schema:
+            $ref: '#/components/schemas/to-do/properties/details'
+      responses:
+        '200':
+          description: A Json object of to-do
+          content:
+            application/json:
+              schema:
+                type: string
+                content:
+                $ref: '#/components/schemas/to-do/properties/details'
+        '404':
+          description: Requested to-do's detail not found
+  /to-dos/{to-doStatus}:
+    patch:
+      summary: Update to-do Status
+      parameters:
+        - name: to-doStatus
+          in: path
+          required: true
+          description: Parameter of to-do that you wanted the update
+          schema:
+            $ref: '#/components/schemas/to-do/properties/status'
+      responses:
+        '200':
+          description: A Json object of to-do
+          content:
+            application/json:
+              schema:
+                type: string
+                content:
+                $ref: '#/components/schemas/to-do/properties/details'
+        '500':
+          description: server error
+    
+components: 
+  schemas:
+    to-do:
+      title: to-do
+      properties:
+        id:
+          type: integer
+          example: 13
+        details:
+          type: string
+          example: Do the swagger doc
+        status:
+          type: string
+          example: done
+    to-do2:
```

Comment:
burada to-do2 gibi isimlendirme kullanmak yerine ToDoBundleResponse ToDoResponse ToDoRequest gibi ayırman lazım isimlendirmelerin açıklayıcı olmalı